### PR TITLE
Push scheduler Result Proxy URL as Web2 secret to SMS

### PR DESCRIPTION
**Pull request by jbern0rd**

### CODE REVIEW

File: src/main/java/com/iexec/core/sms/SmsService.java [hunk 1]
```diff
@@ -16,35 +16,59 @@
...
+                      final SignatureService signatureService,
+                      final SmsClientProvider smsClientProvider) {
+        this.iexecHubService = iexecHubService;
         this.registryConfiguration = registryConfiguration;
+        this.resultRepositoryConfiguration = resultRepositoryConfiguration;
         this.signatureService = signatureService;
         this.smsClientProvider = smsClientProvider;
     }
 
+    private Optional<String> getVerifiedSmsUrl(String chainTaskId) {
```
Comment (nabil-Tounarti) [inline src/main/java/com/iexec/core/sms/SmsService.java:63]: Wouldn't it be better to add final to the chainTaskId parameter for consistency? The same applies to chainTaskId and tag parameters in the getVerifiedSmsUrl function.

Comment (jbern0rd) [reply src/main/java/com/iexec/core/sms/SmsService.java]: Thanks, fixed in 174bed6

File: src/main/java/com/iexec/core/sms/SmsService.java [hunk 2]
```diff
@@ -147,4 +170,51 @@ Optional<String> generateEnclaveChallenge(String chainTaskId, String smsUrl) {
...
+    }
+
+    /**
+     * Pushes worker pool default Result Proxy URL to SMS.
+     *
+     * @param event The default result-proxy URL.
+     */
+    @EventListener
+    public void pushWorkerpoolResultProxyUrl(final TaskInitializedEvent event) {
+        log.debug("pushWorkerpoolResultProxyUrl");
```
Comment (Natchica) [inline src/main/java/com/iexec/core/sms/SmsService.java:199]: Is this there on purpose (if yes, shouldn't it be a bit more descriptive ?) or did you forget to delete it ?
```suggestion
        log.debug("Call to pushWorkerpoolResultProxyUrl(event) [event: {}]", event);
```

Comment (jbern0rd) [reply src/main/java/com/iexec/core/sms/SmsService.java:199]: This is an habit of mine, having a debug log message at the beginning of methods to see how methods are chained.
I don't really like the **Call to** prefix which will be added in many places for little added-value.
I will try the end of your suggestion to see the produced output.

Comment (jbern0rd) [reply src/main/java/com/iexec/core/sms/SmsService.java]: see 4af05bf

**Review by nabil-Tounarti — APPROVED** [review]

**Review by thewhitewizard — APPROVED** [review]

**Pull request merged**